internal/db/db_trace.go

```diff
@@ -11,10 +12,37 @@ import (
 type dbTracer struct{}
 
 func (dt *dbTracer) TraceQueryStart(ctx context.Context, conn *pgx.Conn, data pgx.TraceQueryStartData) context.Context {
-	logrus.WithContext(ctx).Debugf("Executing SQL with args %v: %s", data.Args, data.SQL)
+	logrus.Debug(formatSqlLog(data))
 	return ctx
 }
 
 func (dt *dbTracer) TraceQueryEnd(ctx context.Context, conn *pgx.Conn, data pgx.TraceQueryEndData) {
 	// no-op
 }
+
+func formatSqlLog(data pgx.TraceQueryStartData) string {
+	if logrus.GetLevel() > logrus.DebugLevel {
+		return ""
+	}
+
+	d := make([]interface{}, len(data.Args))
+	copy(d, data.Args)
+	for i, v := range d {
+		if b, ok := v.([]byte); ok {
```

Comment:
So the thing it's not a `[]byte` it's a `json.RawMessage` :

I was messing around here:
https://github.com/osbuild/image-builder/commit/d31d61c7c773b898bfa324ad1c5f51aeebba8a51

Reply:
@kingsleyzissou and I verified that the correct logs present  sudo docker compose logs backend | grep -i 'insert'  
image-builder-backend-1  | time="2024-03-21T12:22:53Z" level=debug msg="Executing SQL: \n\t\tINSERT INTO composes(job_id, request, created_at, account_number, email, org_id, image_name, client_id, blueprint_version_id)\n\t\tVALUES ($1, $2, CURRENT_TIMESTAMP, $3, $4, $5, $6, $7, $8); args: [03120b06-71a4-429a-bc40-03c9922272a1 {\"client_id\":\"ui\"... 6089719 insights-qe-stage-notifications+insights-qa@redhat.com 11789772 <nil> 0x4001792110 <nil>]" func="github.com/osbuild/image-builder/internal/db.(*dbTracer).TraceQueryStart" file="/go/src/github.com/osbuild/image-builder/internal/db/db_trace.go:16"

Reply:
The `json.RawMessage` contains the user email address etc, are we okay with logging all of that information?

Reply:
Makes sense. Yeah we will probably need to add a case for converting the `json.RawMessage` too in that case :)

Reply:
Ah that is right, thanks! Added a test case for this too.